ncm-grub/src/main/perl/grub.pm

```diff
@@ -613,6 +701,89 @@ sub get_info
     return \@entries;
 };
 
+=item current_arguments
+
+Get the current arguments. Return current arguments as string and as parsed hasref
+
+C<track> option is passed to C<convert_grubby_arguments>.
+
+=cut
+
+sub get_current_arguments
+{
+    my ($self, $default, $track) = @_;
+
+    my $entries = $self->get_info($default);
+    if (scalar @$entries > 1) {
+        $self->warn("More than one grub entry for kernel $default found.",
+                    " Only first entry / lowest index will be modified");
+    };
+
+    # Check current arguments
+    my $current = '';
+    if (@$entries && $entries->[0]->{args} && $entries->[0]->{args} =~ m/^\"(.*)\"$/) {
+        $current = $1;
+        $self->verbose("found current args for kernel $default: '$current'");
+    }
+
+    my $currargs = $self->convert_grubby_arguments($current, undef, $track);
+    my @cremove = sort keys %{$currargs->{remove}};
+    if (@cremove) {
+        $self->error("Arguments to remove '@cremove' found in current '$current', must be error in parser");
+    };
+
+    return $current, $currargs;
+}
+
+=item sanitize_arguments
+
+Sanitize the current arguments
+
+=cut
+
+sub sanitize_arguments
+{
+    my ($self, $default) = @_;
+
+    my ($current, $currargs) = $self->get_current_arguments($default, 1);
+
+    my $add = {};
+    my $remove = {};
+
+    # Not caring about remove, these are faulty anyway
+    # For all add that have more than one value
+    foreach my $name (sort keys %{$currargs->{add}}) {
+        my $values = $currargs->{add}->{$name};
+        my $len = scalar(@$values);
+        if ($len > 1) {
+            $self->info("Found $len values for $name, replacing with last one: ",
```

Comment:
just found out why installing latest kernel puts hosts with multiple mdadm arrays in dracut mode..
```2022/05/10-15:51:25 [INFO] Found 3 values for rd.md.uuid, replacing with last one: 8598f4d5:427993ed:e1092e63:748077b5 , 78d92008:5cd1d326:db95a259:9c632937 , 8795753f:47a571d7:410362b8:b05ac2b9```

thi should be possible though, from dracut manual:
```     rd.md.uuid=<md raid uuid>
           only activate the raid sets with the given UUID. This
           parameter can be specified multiple times.```